test/masterzTemplate.ts

```diff
@@ -147,14 +199,31 @@ describe('MasterZTemplate', function () {
       expect(checkpoints[0].state).equals(2) // check checkpoint status
     })
 
+    itIf(integrationTestsOn)('Should not start checkpoint if not admin', async function () {
+      await deployProject()
+      await approveDonationToContract()
+      await (await masterzTemplateContract.connect(owner).donate(0, BigNumber.from('10'))).wait()
```

Comment:
In general, I believe you need to collect connect only when you are switching user, not always, though I might be wrong